lgc/patch/PatchEntryPointMutate.cpp

```diff
@@ -965,9 +965,12 @@ uint64_t PatchEntryPointMutate::generateEntryPointArgTys(ShaderInputs *shaderInp
       *userDataArg.argIndex = argTys.size() + argOffset;
     unsigned dwordSize = userDataArg.argDwordSize;
     if (userDataArg.userDataValue != static_cast<unsigned>(UserDataMapping::Invalid)) {
+      // Most of user data metadata entries is 1 except root push descriptors
```

Comment:
I can't parse this comment.

> Most of user data metadata entries is 1

It's not clear what the unit is here. This could be the number of bytes, the length of each metadata entry, etc.

>  except root push descriptors

nit: except for, nit2: missing `.` at the end of the sentence

>  root push descriptors

I'm not very familiar with this part of the codebase, what are root push descriptors? I've heard of root descriptors and push contacts, but I'm not sure if that's related here or not.

Reply:
Here entries means one  USER_DATA metadata. For root descriptors, it means entries number is equal to the size of the DWORD.

Reply:
Can you add this explanation to the comment?

Reply:
> "pal USER_DATA" metadata (e.g. dump

it's a bit long and introductory, i am not sure how to make it into comment

Reply:
Any more comment?